### PR TITLE
install.sh: check sudo and ask for prefix

### PR DESCRIPTION
**Pull request by dawidd6**
Port: https://github.com/Linuxbrew/install/blob/2cc4266960a388510cff7325d4172e3344d7eecf/install#L212-L217

With this change, the installer asks interactively (not on CI) where to install Homebrew (`/home/linuxbrew/.linuxbrew` with sudo or `~/.linuxbrew` without sudo).

cc @sjackman

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -53,6 +54,19 @@ tty_red="$(tty_mkbold 31)"
 tty_bold="$(tty_mkbold 39)"
 tty_reset="$(tty_escape 0)"
 
+have_sudo() {
```
Comment (MikeMcQuaid) [inline install.sh]: This naming (or `execute_sudo`) feels a bit strange given sudo is probably always available but just not accessible and that this returns true unconditionally on macOS.

File: install.sh [hunk 2]
```diff
@@ -187,6 +206,29 @@ fi
 cd "/usr" || exit 1
 
 ####################################################################### script
+if [[ -n "${HOMEBREW_ON_LINUX-}" ]]; then
+  if [[ -n "${CI-}" ]] || [[ -w "$HOMEBREW_PREFIX_DEFAULT" ]] || [[ -w "/home/linuxbrew" ]] || [[ -w "/home" ]]; then
```
Comment (MikeMcQuaid) [inline install.sh]: What's this for?

Comment (dawidd6) [reply install.sh]: I have no idea, just ported it from `Linuxbrew/install`. @sjackman might know it.

File: install.sh (hunk 1 above)
Comment (dawidd6) [reply install.sh]: `have_sudo_access` might be a better fit here. Thoughts?

Comment (MikeMcQuaid) [reply install.sh]: Perfect!

Comment (MikeMcQuaid) [reply install.sh]: Thoughts: checking for access on macOS too makes sense, I think, and just erroring out if you don't have `sudo` there?

**Comment by MikeMcQuaid** [comment]
Thanks again @dawidd6!

**Comment by sjackman** [comment]
Fantastic! Thanks again, Dawid! 🏆

**Pull request merged**